web-server/v0.4/src/routes/Dashboard/RunComparison.js

```diff
@@ -303,6 +331,126 @@ export default class RunComparison extends ReactJS.Component {
     this.setState({ timeseriesDropdownSelected: timeseriesDropdownSelected });
   };
 
+  savePDF = async () => {
```

Comment:
Since this function call takes advantage of Promises, we can remove the reference to `async` as the usage of Promises ensures an asynchronous behavior in the function call.

Reply:
Resolved.

Reply:
```
savePDF = () => {
    const { defaultChecked, pdfHeader, pdfName } = this.state;
    /* maps through the state array containing the IDs and saves the pdf */

    document.getElementById("table").offsetHeight;
    var doc = new jsPDF('p', 'mm', [1050, 1475], true);
    doc.text(pdfHeader, 172, 10, 'center');
    var position = 5;
    var fullIHeight = 0;
    let selectedElements = [] ;

    defaultChecked.forEach((element) => {
      selectedElements.push(html2canvas(document.querySelector(`#${  element}`)).then((canvas, index) => {
        doc.setPage(index);
        canvas.getContext('2d');
        let imgData = canvas.toDataURL();
        let imgWidth = 345;
        let pageHeight = 510;
        let imgHeight = (canvas.height * imgWidth) / canvas.width;
        position = fullIHeight + 5;
        fullIHeight += imgHeight;
        let heightLeft = fullIHeight;
        doc = doc.addImage(imgData, 'JPEG', 10, position, imgWidth, imgHeight);
        heightLeft -= pageHeight;
        {/*while (heightLeft >= 0) {
          position = heightLeft - fullIHeight;
          doc.addPage();
          doc = doc.addImage(imgData, 'JPEG', 10, position, imgWidth, imgHeight);
          heightLeft -= pageHeight;
        }*/}
      }))
    })

    Promise.all(selectedElements).then(() => {
      doc.save(`${pdfName  }.pdf`);
      this.setState({
        visible: false,
        generatingPdf: false,
      });
    });
  };
```